src/main/java/spireMapOverhaul/zones/brokenSpace/patches/BrokenSpaceRenderPatch.java

```diff
@@ -0,0 +1,196 @@
+        sb.flush();
+        fbo.end();
+
+        TextureRegion region = new TextureRegion(fbo.getColorBufferTexture());
+        region.flip(false, true);
+
+
+        sb.setShader(brokenSpaceShader);
+        sb.setColor(Color.WHITE);
+        brokenSpaceShader.setUniformf("u_time", shaderTimer + timerOffset);
+        brokenSpaceShader.setUniformf("u_strength", strength);
+        brokenSpaceShader.setUniformf("u_chrAb", chrAb);
+
+        sb.draw(region, 0, 0);
+        sb.setShader(null);
+    }
+
+
+    private static boolean shouldRenderBrokenSpaceShader(RewardItem __instance) {
+        return inBrokenSpace() && (__instance.type == RewardItem.RewardType.CARD);
+
+    }
+
+    private static boolean shouldRenderBrokenSpaceShader(AbstractCard __instance) {
+        return inBrokenSpace() && isUnnatural(__instance) && !adp().masterDeck.contains(__instance);
+    }
+
+    private static boolean shouldRenderBrokenSpaceShader(AbstractRelic __instance, boolean isShop) {
+        return inBrokenSpace() && isUnnatural(__instance, isShop) && !adp().relics.contains(__instance);
+    }
+
+    private static boolean isUnnatural(AbstractRelic __instance, boolean isShop) {
+        boolean isUnnatural = __instance.tier == AbstractRelic.RelicTier.SPECIAL || __instance.tier == AbstractRelic.RelicTier.BOSS;
+        if (!isShop) {
+            isUnnatural = isUnnatural || __instance.tier == AbstractRelic.RelicTier.SHOP;
+
+        }
+
+        return isUnnatural;
+    }
+
+
+    private static boolean isUnnatural(AbstractCard __instance) {
+        return __instance.color != adp().getCardColor() && BrokenSpaceZone.UnnaturalCardField.unnatural.get(__instance);
+    }
+
+
+    private static boolean inBrokenSpace() {
+        return ZonePatches.currentZone() != null && ZonePatches.currentZone().id.equals("BrokenSpace");
+    }
+
+
+    @SpirePatch(
+            clz = CardCrawlGame.class,
+            method = "render"
+    )
+    public static class UpdateShaderTimer {
+        @SpirePostfixPatch
+        public static void updateShaderTimer() {
+
+            shaderTimer += Gdx.graphics.getDeltaTime();
+
+        }
+
+    }
+
+    @SpirePatch2(
+            clz = AbstractCard.class,
+            method = SpirePatch.CLASS
+    )
+    public static class RandomTimeField {
+        public static SpireField<Float> randomTime = new SpireField<>(() -> (float) Math.random() * 1000f);
+    }
+
+
+
+    static {
+        brokenSpaceShader = new ShaderProgram(Gdx.files.internal(makeShaderPath("BrokenSpace/Glitch.vs")), Gdx.files.internal(makeShaderPath( "BrokenSpace/Glitch.fs")));
+        if (!brokenSpaceShader.isCompiled()) {
+            logger.warning("Broken Space shader: " + brokenSpaceShader.getLog());
+        }
+        brokenSpaceShader.begin();
+
+
+        fbo = new FrameBuffer(Pixmap.Format.RGBA8888, Gdx.graphics.getWidth(), Gdx.graphics.getHeight(), false, false);
```

Comment:
This framebuffer size might be a bit overkill? Not that good with framebuffers and shaders so maybe i'm wrong, but generally making a fullscreen framebuffer is not very optimal

Reply:
I think otherwise I would have to manually reposition the render locations. I may be wrong but this is certainly the simplest path.